lib/dfdutils/createdfd.c

```diff
@@ -158,6 +162,10 @@ static void writeSample(uint32_t *DFD, int sampleNo, int channel,
         upper.f = 1.0f;
         lower.f = 0.0f;
         break;
+    case s_S10_5:
+        assert(bits == 16 && "Format with this suffix must be 16 bits per channel.");
+        upper.i = 32;
+        lower.i = ~upper.i + 1; // -32
```

Comment:
This doesn't seem correct at all.

Per the Vulkan spec, the S10_5_NV format has a range of [-1024,1023.96875] in floating point format, as it's a fixed-point format with 1 sign bit, 10 integer bits and 5 fractional bits.

Reply:
Thus, assuming this is treated as an integer format, then the range should be the full 0 - 0xFFFFFFFF range, and if it is treated as float, then it probably should be -1024.0 - 1023.96875.

Reply:
Float flag is not set.